src/store/models/network.ts

```diff
@@ -943,6 +978,79 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  renameNode: thunk(
+    async (actions, { node, newName }, { getState, getStoreActions, injections }) => {
+      if (!newName) throw new Error(l('renameErr', { newName }));
+
+      if (node.status === Status.Started) {
+        await actions.stop(node.networkId);
+      }
+
+      const networks = getState().networks;
+      const network = networks.find(n => n.id === node.networkId);
+      if (!network) throw new Error(l('networkByIdErr', { networkId: node.networkId }));
+
+      let updatedNode;
+      switch (node.type) {
+        case 'lightning':
+          updatedNode = network?.nodes.lightning.find(n => n.id === node.id) as Node;
```

Comment:
I think it makes sense to move this `switch` logic into the [utils/network](https://github.com/jamaljsr/polar/blob/master/src/utils/network.ts) file since that is responsible for manipulating the network object and setting the file paths. You could just have `renameNode(network, node, newName) which returns the updated `network` with the node's name and path changed.

Reply:
Doing triggered this error:

> Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
> 1. You might have mismatching versions of React and the renderer (such as React DOM)
> 2. You might be breaking the Rules of Hooks
> 3. You might have more than one copy of React in the same app

This is probably because the `switch` has a `store` call in it, I am not sure though. I have maintained it in the `models/network.ts`  file for now and await your feedback

Reply:
The store call is identical in each `case` block. You can remove that from the `switch` and do it once after calling the network util function.